stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/global/GlobalMutableState.kt

```diff
@@ -57,6 +57,18 @@ internal class GlobalMutableState : GlobalState {
     override fun isInitialized(): Boolean {
         return _initialized.value
     }
+
+    internal companion object {
+        private var instance: GlobalMutableState? = null
+
+        internal fun getOrCreate(): GlobalMutableState {
+            return instance ?: GlobalMutableState().also { globalState ->
+                instance = globalState
+            }
+        }
+
+        internal fun create(): GlobalMutableState = GlobalMutableState()
```

Comment:
We may want a method to force the creation of a new GlobalState. In our case, we use it in our unit tests, otherwise the state will be shared between tests.